src/Vesting.ts

```diff
@@ -0,0 +1,207 @@
+import { Address, Datum, TxOutput, Value} from "@hyperionbt/helios";
+import { StellarContract, redeem, datum, txn } from "../lib/StellarContract.js";
+import contract from "./Vesting.hl";
+import { StellarTxnContext } from "../lib/StellarTxnContext";
+
+import {ADA} from "../lib/StellarTestHelper";
+
+export type VestingParams = {
+    sponsor: WalletEmulator;
+    payee: Address;
+    deadline: bigint;
+};
+
+export type VestingDatumArgs = {
+    sponsor: PubKeyHash;
+    payee: PubKeyHash;
+    time: number | bigint;
+};
+
+export class Vesting extends StellarContract<VestingParams> {
+    contractSource() {
+        return contract;
+    }
+    @datum
+    mkDatum({
+        sponsor,
+        payee,
+	time
+    }: VestingDatumArgs): InlineDatum {
+        //!!! todo: make it possible to type these datum helpers more strongly
+        const t = new this.configuredContract.types.Datum(
+            sponsor.bytes,
+	    payee.bytes,
+            time
+        );
+        return Datum.inline(t._toUplcData());
+    }
+    @txn
+    async mkTxnDepositValueForVesting(
+        { sponsor, payee, deadline }: VestingParams,
+        tcx: StellarTxnContext = new StellarTxnContext()
+    ): Promise<StellarTxnContext | never> {
+	    // so far value is hardcoded: 
+	    	const margin = 1n * ADA;
```

Comment:
changing this line switches between `unexpected` and `not enough ada to cover tx fees`